--- a/sample_apps/subscribe_publish_sample/aws_iot_config.h	2016-03-03 05:08:09.851915828 -0800
+++ b/sample_apps/subscribe_publish_sample/aws_iot_config.h	2016-03-03 05:14:37.428987965 -0800
@@ -23,34 +23,78 @@
 
 // Get from console
 // =================================================
+#ifndef AWS_IOT_MQTT_HOST
 #define AWS_IOT_MQTT_HOST              "" ///< Customer specific MQTT HOST. The same will be used for Thing Shadow
+#endif
+#ifndef AWS_IOT_MQTT_PORT
 #define AWS_IOT_MQTT_PORT              8883 ///< default port for MQTT/S
+#endif
+#ifndef AWS_IOT_MQTT_CLIENT_ID
 #define AWS_IOT_MQTT_CLIENT_ID         "c-sdk-client-id" ///< MQTT client ID should be unique for every device
+#endif
+#ifndef AWS_IOT_MY_THING_NAME
 #define AWS_IOT_MY_THING_NAME 		   "AWS-IoT-C-SDK" ///< Thing Name of the Shadow this device is associated with
+#endif
+#ifndef AWS_IOT_ROOT_CA_FILENAME
 #define AWS_IOT_ROOT_CA_FILENAME       "aws-iot-rootCA.crt" ///< Root CA file name
+#endif
+#ifndef AWS_IOT_CERTIFICATE_FILENAME
 #define AWS_IOT_CERTIFICATE_FILENAME   "cert.pem" ///< device signed certificate file name
+#endif
+#ifndef AWS_IOT_PRIVATE_KEY_FILENAME
 #define AWS_IOT_PRIVATE_KEY_FILENAME   "privkey.pem" ///< Device private key filename
+#endif
 // =================================================
 
 // MQTT PubSub
+#ifndef AWS_IOT_MQTT_TX_BUF_LEN 
 #define AWS_IOT_MQTT_TX_BUF_LEN 512 ///< Any time a message is sent out through the MQTT layer. The message is copied into this buffer anytime a publish is done. This will also be used in the case of Thing Shadow
+#endif
+#ifndef AWS_IOT_MQTT_RX_BUF_LEN
 #define AWS_IOT_MQTT_RX_BUF_LEN 512 ///< Any message that comes into the device should be less than this buffer size. If a received message is bigger than this buffer size the message will be dropped.
+#endif
+#ifndef AWS_IOT_MQTT_NUM_SUBSCRIBE_HANDLERS
 #define AWS_IOT_MQTT_NUM_SUBSCRIBE_HANDLERS 5 ///< Maximum number of topic filters the MQTT client can handle at any given time. This should be increased appropriately when using Thing Shadow
+#endif
 
 // Thing Shadow specific configs
+#ifndef SHADOW_MAX_SIZE_OF_RX_BUFFER 
 #define SHADOW_MAX_SIZE_OF_RX_BUFFER AWS_IOT_MQTT_RX_BUF_LEN+1 ///< Maximum size of the SHADOW buffer to store the received Shadow message
+#endif
+#ifndef MAX_SIZE_OF_UNIQUE_CLIENT_ID_BYTES
 #define MAX_SIZE_OF_UNIQUE_CLIENT_ID_BYTES 80  ///< Maximum size of the Unique Client Id. For More info on the Client Id refer \ref response "Acknowledgments"
+#endif
+#ifndef MAX_SIZE_CLIENT_ID_WITH_SEQUENCE
 #define MAX_SIZE_CLIENT_ID_WITH_SEQUENCE MAX_SIZE_OF_UNIQUE_CLIENT_ID_BYTES + 10 ///< This is size of the extra sequence number that will be appended to the Unique client Id
+#endif
+#ifndef MAX_SIZE_CLIENT_TOKEN_CLIENT_SEQUENCE
 #define MAX_SIZE_CLIENT_TOKEN_CLIENT_SEQUENCE MAX_SIZE_CLIENT_ID_WITH_SEQUENCE + 20 ///< This is size of the the total clientToken key and value pair in the JSON
+#endif
+#ifndef MAX_ACKS_TO_COMEIN_AT_ANY_GIVEN_TIME
 #define MAX_ACKS_TO_COMEIN_AT_ANY_GIVEN_TIME 10 ///< At Any given time we will wait for this many responses. This will correlate to the rate at which the shadow actions are requested
+#endif
+#ifndef MAX_THINGNAME_HANDLED_AT_ANY_GIVEN_TIME
 #define MAX_THINGNAME_HANDLED_AT_ANY_GIVEN_TIME 10 ///< We could perform shadow action on any thing Name and this is maximum Thing Names we can act on at any given time
+#endif
+#ifndef MAX_JSON_TOKEN_EXPECTED
 #define MAX_JSON_TOKEN_EXPECTED 120 ///< These are the max tokens that is expected to be in the Shadow JSON document. Include the metadata that gets published
+#endif
+#ifndef MAX_SHADOW_TOPIC_LENGTH_WITHOUT_THINGNAME
 #define MAX_SHADOW_TOPIC_LENGTH_WITHOUT_THINGNAME 60 ///< All shadow actions have to be published or subscribed to a topic which is of the format $aws/things/{thingName}/shadow/update/accepted. This refers to the size of the topic without the Thing Name
+#endif
+#ifndef MAX_SIZE_OF_THING_NAME
 #define MAX_SIZE_OF_THING_NAME 20 ///< The Thing Name should not be bigger than this value. Modify this if the Thing Name needs to be bigger
+#endif
+#ifndef MAX_SHADOW_TOPIC_LENGTH_BYTES
 #define MAX_SHADOW_TOPIC_LENGTH_BYTES MAX_SHADOW_TOPIC_LENGTH_WITHOUT_THINGNAME + MAX_SIZE_OF_THING_NAME ///< This size includes the length of topic with Thing Name
+#endif
 
 // Auto Reconnect specific config
+#ifndef AWS_IOT_MQTT_MIN_RECONNECT_WAIT_INTERVAL
 #define AWS_IOT_MQTT_MIN_RECONNECT_WAIT_INTERVAL 1000 ///< Minimum time before the First reconnect attempt is made as part of the exponential back-off algorithm
+#endif
+#ifndef AWS_IOT_MQTT_MAX_RECONNECT_WAIT_INTERVAL
 #define AWS_IOT_MQTT_MAX_RECONNECT_WAIT_INTERVAL 8000 ///< Maximum time interval after which exponential back-off will stop attempting to reconnect.
+#endif
 
 #endif /* SRC_SHADOW_IOT_SHADOW_CONFIG_H_ */
