--- a/feeds/packages/net/wget/Makefile	2016-01-21 18:00:27.000000000 -0800
+++ b/feeds/packages/net/wget/Makefile	2016-03-25 21:33:52.395209702 -0700
@@ -25,7 +25,7 @@
 define Package/wget/Default
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+libpcre
+  DEPENDS:=+libpcre +zlib
   SUBMENU:=File Transfer
   TITLE:=Non-interactive network downloader
   URL:=http://www.gnu.org/software/wget/index.html
